### ISSUE TITLE
[Feature request] check all places where there are activity-id's

### ISSUE THREAD

**Issue by matmaxgeds** [codeforIATI/activity-id-checker#2]
I think the checker currently looks just in the main activity-id field, but it would also be useful to find/report conflicts in the provider/recipient activity, and the @ref field used for related activities

**Comment by stevieflow** [comment]
@matmaxgeds my understanding of this checking is that the list of ``iati-identfier`` is currently populated from the ``iati-actvity``.  This makes sense (to me!) in this context, as we just want to query the list of *known IATI activity identifiers* that are published.

I understand what you mean about other places, but I think initially we are *just* populating a list of the 1million+ activities.

**Comment by matmaxgeds** [comment]
@stevieflow agreed, I thought it would also be useful for the tool to be able to check the linked but different question of 'show me everywhere this activity-id is in use'.

I also wonder if there is a (not handleable by us I think) historical aspect here e.g. how to tell if you are using an activity-id that has been used in the past, that would also be an edge case that could break things. I do think we should think about a way to store historical IATI data with a field that says whether it is live now.
